io/eolearn/io/geometry_io.py

```diff
@@ -90,143 +143,101 @@ def dataset_crs(self):
 
-            extent = gpd.GeoSeries([bbox.geometry], crs=bbox_crs)
-            vectors = gpd.clip(vectors, extent, keep_geom_type=True)
 
-        return vectors
+class GeopediaVectorImportTask(_BaseVectorImportTask):
+    """ A task for importing `Geopedia <https://geopedia.world>`__ features into EOPatch vector features
+    """
 
-    def execute(self, eopatch=None, *, bbox=None):
+    def __init__(self, feature, geopedia_table, reproject=True, clip=False, **kwargs):
         """
-        :param eopatch: An existing EOPatch. If none is provided it will create a new one.
-        :type eopatch: EOPatch
-        :param bbox: A bounding box for which to load data. By default, if none is provided, it will take a bounding box
-            of given EOPatch. If given EOPatch is not provided it will load the entire dataset.
-        :return: An EOPatch with an additional vector feature
-        :rtype: EOPatch
+        :param feature: A vector feature into which to import data
+        :type feature: (FeatureType, str)
+        :param geopedia_table: A Geopedia table from which to retrieve features
+        :type geopedia_table: str or int
+        :param reproject: Should the geometries be transformed to coordinate reference system of the requested bbox?
+        :type reproject: bool, default = True
+        :param clip: Should the geometries be clipped to the requested bbox, or should be geometries kept as they are?
+        :type clip: bool, default = False
+        :param kwargs: Additional args that will be passed to `GeopediaFeatureIterator`
         """
-        eopatch = eopatch or EOPatch()
-        bbox = bbox or eopatch.bbox
-
-        if not eopatch.bbox:
-            eopatch.bbox = bbox
-
-        prepared_bbox = self._prepare_bbox(bbox)
-        vectors = self._load_vector_data(prepared_bbox)
-
-        eopatch[self.feature] = self._reproject_and_clip(vectors, eopatch.bbox)
-
-        if not eopatch.bbox:
-            eopatch.bbox = BBox(list(eopatch[self.feature].total_bounds), crs=eopatch[self.feature].crs.to_epsg())
-
-        return eopatch
-
-
-class GeopediaVectorImportTask(VectorImportTask):
-    """ A task for importing Geopedia features into EOPatch vector features
-    """
-    def __init__(self, feature, geopedia_table, reproject=True, clip=False, config=None, **kwargs):
         self.geopedia_table = geopedia_table
-        super().__init__(feature=feature, path=None, reproject=reproject, clip=clip, config=config, **kwargs)
-
-    def _prepare_bbox(self, bbox):
-        """ This makes sure that the bounding box is in correct CRS. The final bounds can be larger than given bbox
-        """
-        if bbox is None:
-            self._dataset_crs = CRS.WGS84  # geopedia always returns data in WGS84
-            return None
-
-        self._dataset_crs = bbox.crs
-        return bbox.transform_bounds(CRS.POP_WEB)
+        self.dataset_crs = CRS.WGS84
```

Comment:
`self.dataset_crs` is actually not needed in this task at all. You already read CRS from responses.

Reply:
True.